### PR TITLE
Add ability to configure toolbar (block and inline styles)

### PR DESCRIPTION
**Pull request by roy-gbiv**
Hello! Would love to be able to use this (it's so purdy; the linking UX is fantastic), enough so that I went ahead and added the one piece of functionality missing that I would need for my project. :) 

I wanted to be able to limit what modification options are presented to my users (i.e. only bold, italics, list, and linking). Please let me know of any suggestions you'd have and I'll be sure to get them in tomorrow. Would love to be able to use this by Friday.

### CODE REVIEW

**Comment by brijeshb42** [comment]
Hey, I like what you want as the feature. But I am not in favour of introducing a dependency that is not used heavily throughout the project. In this case, `lodash`. Besides, what you are trying to do here can be done easily without actually changing `medium-draft`. Also, right now, the toolbar options can be changed any time. For ex - initially you can show 3 buttons, but based on some user action or in general at a later time, you can add/remove buttons as you want. But in your case, since you are calculating buttons inside the constructor, you will not be able to add/remove button at a later time passing a different `toolbarConfig` as prop.

Here is how I would do what you want without touching `medium-draft` itself.

```js
import _ from 'lodash';
import Editor from 'medium-draft';
import { BLOCK_BUTTONS, INLINE_BUTTONS } from 'medium-draft/lib/components/toolbar';

class App extends React.component {

  constructor(props) {
    super(props);

    const toolbarConfig = {
      block: [
        'ordered-list-item',
        'unordered-list-item',
        'blockquote',
        'header-three',
        'todo',
      ],
      inline: [
        'BOLD',
        'ITALIC',
        'UNDERLINE',
        'hyperlink',
        'HIGHLIGHT',
      ],
    };
    this.blockButtons = this.configureToolbarBlockOptions (toolbarConfig);
    this.inlineButtons = this.configureToolbarInlineOptions (toolbarConfig);
  }

  configureToolbarInlineOptions() {
    // your definition here
  }

  configureToolbarBlockOptions() {
    // your definition here
  }

  // ...other methods

  render () {
    return (
      <Editor
        {...otherProps}
        blockButtons={this.blockButtons}
        inlineButtons={this.inlineButtons}
      />
    );
  }
}
```

**Comment by roy-gbiv** [comment]
This sounds great. I would much rather not touch medium-draft itself, but I did not have documentation specifying that capability. I'll see if I can work that out, and then close / cancel the PR, and ask questions here if they come up. Thank you!

**Comment by roy-gbiv** [comment]
Actually, if the lodash was truly the only problem, then doing without lodash would be easy. I can make that change. :)  

Thank you for the proposed solution on how to change the toolbar. :) However, does it not seem a bit dangerous to be reaching into a lib file in this way? After all, another developer could easily change the form of BLOCK_BUTTONS / INLINE_BUTTONS through something as big as moving those constants or as small as changing the the INLINE_BUTTON styles to all lower case, which would immediately break my app. Or they could be outright moved.

**Comment by roy-gbiv** [comment]
You make a good point regarding not being able to change the toolbarConfig after the constructor, though, so I'll go ahead and move it out of the constructor as well. 👍

**Comment by roy-gbiv** [comment]
Updated. Should now be able to update the toolbar at render and no longer uses `lodash`. Let me know if there's anything else I can do. :) 

I looked into how lib/ directory is used in React (as I had only knew of how it was used for Ruby on Rails and that does not seem to be the way it is used in this project). I am hoping this solution is preferable.

File: package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
   "dependencies": {
     "draft-js": "^0.10.0",
     "immutable": "^3.7.6",
+    "lodash": "^4.17.4",
```
Comment (roy-gbiv) [inline package.json]: Oops, didn't commit the removal of this change.

**Comment by roy-gbiv** [comment]
Apologies. Git mixups sent my old code back again. O_O'

**Comment by roy-gbiv** [comment]
Squashing into single commit was not working out.

**Comment by brijeshb42** [comment]
> Thank you for the proposed solution on how to change the toolbar. :) However, does it not seem a bit dangerous to be reaching into a lib file in this way? After all, another developer could easily change the form of BLOCK_BUTTONS / INLINE_BUTTONS through something as big as moving those constants or as small as changing the the INLINE_BUTTON styles to all lower case, which would immediately break my app.

I am trying to follow `semver` approach. So there is almost negligible chance that there will be any kind of breaking changes like what you are talking about. And I am the only developer responsible for maintaining this project. Others might open a PR once in a while, but I am the one who will merge them and I make certain that there are no breaking changes

If at a later time, breaking changes are to be introduced, there will be a deprecation warning first before actually doing that kind of thing.

**Comment by roy-gbiv** [comment]
Ah, I see. Well, thank you for your work!

**Pull request closed without merging**